sqltypes/json.go

```diff
@@ -0,0 +1,35 @@
+package sqltypes
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+
+	"github.com/upfluence/errors"
+)
+
+type JSONValue[T any] struct {
+	Data  T
+	Valid bool
+}
+
+func (jv *JSONValue[T]) Scan(v interface{}) error {
+	switch vv := v.(type) {
+	case []byte:
+		return json.Unmarshal(vv, &jv.Data)
+	case string:
+		return json.Unmarshal([]byte(vv), &jv.Data)
+	case nil:
+		jv.Data = *new(T)
```

Comment:
I've putted it in case the same object is reused through different `Scan`. that way you won't have the previous value remaining if the value of the current record is null.